compiler/qsc_partial_eval/src/lib.rs

```diff
@@ -1108,9 +1113,18 @@ impl<'a> PartialEvaluator<'a> {
         store_item_id: StoreItemId,
         callable_decl: &CallableDecl,
         args_value: Value,
-    ) -> Value {
+        callee_expr_span: Span,
+    ) -> Result<Value, Error> {
         // Intrinsic callables that make it to this point are expected to be unitary.
-        assert_eq!(callable_decl.output, Ty::UNIT);
+        if callable_decl.output != Ty::UNIT {
+            return Err(Error::Unexpected(
+                format!(
+                    "dynamic call to non-Unit intrinsic `{}`",
```

Comment:
```suggestion
                    "non-classical call to non-Unit intrinsic `{}`",
```

Reply:
This is a small nit, feel free to ignore :)

Reply:
fixed!